### PR TITLE
Match certificate tags to esloader

### PR DESCRIPTION
**Pull request by dadrian**
Some tags are not currently possible (validation type, untrusted, and
self-signed). These tags require peeking into "parsed".

### CODE REVIEW

File: src/certificates.cc [hunk 1]
```diff
@@ -12,12 +12,63 @@
...
 
+namespace {
+
+// TODO: Write function to apply a predicate to every CTServerStatus or
+// RootStoreStatus, with the goal of reducing duplicated code in the
+// certificate_has_X functions.
+
+}  // namespace
+
+std::string translate_certificate_source(int source) {
```
Comment (zakird) [inline src/certificates.cc]: This should probably have everything from https://github.com/Censys/censys-definitions/blob/master/proto/certificate.proto#L18 in it.

File: src/certificates.cc [hunk 2]
```diff
@@ -40,4 +91,202 @@ void expire_status(zsearch::RootStoreStatus* expired) {
...
+            continue;
+        }
+        out->insert(translate_certificate_type(cert_type));
+    }
+    return;
+}
+
+bool certificate_has_ccadb(const zsearch::Certificate& c) {
+    const zsearch::MozillaSalesForceStatus& ccadb = c.audit().mozilla();
+    return ccadb.was_in() || ccadb.current_in();
```
Comment (dadrian) [inline src/certificates.cc]: @zakird this is really the only thing I want review on

File: src/fastjson.cc [hunk 3]
```diff
@@ -337,18 +293,76 @@ void fast_dump_certificate(std::ostream& f,
...
+    // Parseability
+    if (c.post_processed() && c.parsed().empty()) {
+        tags.insert("unparseable");
+    }
+
+    // TODO: Add validation level (DV, OV, EV, etc.). This is currently only
+    // accessible by parsing "parsed".
+
+    // TODO: Add self-signed vs untrusted. Self-signed is currently only
+    // accessible by parsing "parsed".
```
Comment (dadrian) [inline src/fastjson.cc]: @zakird noting these. Not sure what to do about them.

File: src/certificates.cc (hunk 2 above)
Comment (zakird) [inline src/certificates.cc]: Yeah this is fine.

File: src/fastjson.cc (hunk 3 above)
Comment (zakird) [inline src/fastjson.cc]: I agree, a problem for later. Probably need to go into the protobuf. :|

**Pull request merged**